tests/vmi_hook_sidecar_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("HookSidecars", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	var vmi *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		vmi = tests.NewRandomVMIWithEphemeralDisk(tests.RegistryDiskFor(tests.RegistryDiskAlpine))
+		vmi.ObjectMeta.Annotations = map[string]string{
+			"hooks.kubevirt.io/hookSidecars":              `[{"image": "registry:5000/kubevirt/example-hook-sidecar:devel"}]`,
```

Comment:
The image name needs to be dynamically built based on some vars. Take a look at the tests/utills.go KubeVirtRepoPrefix and KubeVirtVersionTag variables and how they are used.

Reply:
@davidvossel  please see the last commit. I changed it to use those constants, but with the change it fails with timeout. Any idea what could be the issue? I keep example-hook-sidecar in cmd/ not in images/, maybe that?

```
HookSidecars
/root/go/src/kubevirt.io/kubevirt/tests/vmi_hook_sidecar_test.go:40
  VMI definition
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_hook_sidecar_test.go:59
    with SM BIOS hook sidecar
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_hook_sidecar_test.go:60
      should successfully start with hook sidecar annotation [It]
      /root/go/src/kubevirt.io/kubevirt/tests/vmi_hook_sidecar_test.go:61

      Timed out after 90.181s.
      Expected
          <bool>: false
      to equal
          <bool>: true

      /root/go/src/kubevirt.io/kubevirt/tests/utils.go:1030
```

Reply:
> please see the last commit.

did you push your changes. I don't see them.

Reply:
I'm sorry, pushed in a wrong branch. Should be here now.